### PR TITLE
Phx 130 loading component

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
PHX-130

### Changes:
Loading skeleton component

### Screenshots:
https://linear.app/bes/issue/PHX-130#comment-4c6c2f91

### CODE REVIEW

File: packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js [hunk 1]
```diff
@@ -17,6 +21,10 @@ const FlexSpaceBetween = styled(FlexSpaceBetweenCenter)`
   align-items: flex-start;
 `;
 
+const SkeletonHeader = styled(FlexCenter)`
```
Comment (tcaiger) [inline packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js:24]: You shouldn't need a custom component to add padding to this. FlexCenter extends [Material UI Box](https://v4.mui.com/components/box/#box) component and it is possible to use that to set the margin.
eg `<FlexCenter mt="10px">`

File: packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js [hunk 2]
```diff
@@ -100,28 +117,42 @@ export const TitleAndDatePickerComponent = ({
...
-            startDate={startDate}
-            endDate={endDate}
-            min={datePickerLimits.startDate}
-            max={datePickerLimits.endDate}
-            onSetDates={updateMeasurePeriod}
-            isLoading={isMeasureLoading}
-          />
-        </MeasureDatePicker>
+      {isMeasureLoading ? (
+        <SkeletonHeader mt="10px">
```
Comment (tcaiger) [inline packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js:121]: The loading section appears to be a different height to the content section and so the side bar moves up and down when it starts and stops loading. I think adding 10px margin bottom will fix it.

File: packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js [hunk 3]
```diff
@@ -54,6 +62,15 @@ const Pin = styled(PinBase)`
   margin: 3px 12px 0 7px;
 `;
 
+const CircularProgress = styled(MuiCircularProgress)`
+  color: #2196f3;
+`;
+
+const Skeleton = styled(MuiSkeleton)`
+  margin-left: ${({ ml = 0 }) => ml}px;
+  margin-top: ${({ mt = 0 }) => mt}px;
```
Comment (tcaiger) [inline packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js:71]:
```suggestion
  margin-top: ${({ mt = 0 }) => mt}px;
  background: none;
```
To make the fading effect from the design, you can turn off the background color on the skeleton.

**Review by tcaiger — CHANGES_REQUESTED** [review]
Hey Biao, this looks cool with the skeleton loader. I just left a suggestion to match the height of the loader to the content but apart from that it looks good.

File: packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js (hunk 1 above)
Comment (billli0) [reply packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js]: Nice, will remove `SkeletonHeader`.

File: packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js (hunk 2 above)
Comment (billli0) [reply packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js]: Sounds good, these component have their own padding and it costs me some times to fix it. Is it ok to use 'min-height' to make them consistent?

File: packages/web-frontend/src/containers/MapOverlayBar/Content.js [hunk 4]
```diff
@@ -6,14 +6,15 @@ export const Content = styled.div`
...
+  padding: 21px 16px 15px 16px;
   color: #ffffff;
 `;
 
 export const ContentText = styled.div`
   font-weight: 500;
   font-size: 16px;
-  line-height: 19px;
+  line-height: 21px;
+  padding-right: 15px;
```
Comment (billli0) [inline packages/web-frontend/src/containers/MapOverlayBar/Content.js:17]: As discussed with David, we want to add a padding between the overlay name and switch button, see screenshot: https://linear.app/bes/issue/PHX-130#comment-3097748d

File: packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js (hunk 2 above)
Comment (tcaiger) [reply packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js]: yep no problem

**Pull request merged**